client/my-sites/plans-features-main/components/free-plan-paid-domain-dialog.tsx

```diff
@@ -160,12 +162,14 @@ function SuggestedPlanSection( {
 
 type DomainPlanDialogProps = {
 	paidDomainName: string;
-	wpcomFreeDomainSuggestion: SingleFreeDomainSuggestion;
+	wpcomFreeDomainSuggestion: DataResponse< DomainSuggestion >;
 	suggestedPlanSlug: PlanSlug;
 	onFreePlanSelected: () => void;
 	onPlanSelected: () => void;
 };
 
+const MODAL_VIEW_EVENT_NAME = 'calypso_free_plan_paid_domain_modal_view';
```

Comment:
Not related to this PR but is there a way for us  to change this event name to `plan_grid_paid_plan_upsell`
CC - @aaronyan 

This is because we are now showing a modal upsell for free plans as well.

Reply:
Updated it as `calypso_plan_upsell_modal_view` by https://github.com/Automattic/wp-calypso/pull/79279/commits/a9b5e8c1e9ac0cf1cb2c4daea2fa8fec4f5bba70 .